src/VisualStudio/Core/Def/SolutionEventMonitor.cs

```diff
@@ -19,42 +19,25 @@ internal sealed class SolutionEventMonitor : IDisposable
         private const string SolutionOpening = "Solution Opening";
         private const string SolutionClosing = "Solution Closing";
 
-        private readonly UIContext? _solutionClosingContext;
-        private IGlobalOperationNotificationService? _notificationService;
+        private readonly UIContext _solutionClosingContext = UIContext.FromUIContextGuid(VSConstants.UICONTEXT.SolutionClosing_guid);
```

Comment:
Just to confirm this is safe to call on any thread at this point? I know in the past there's been some gotchas with this method since it'd implicitly call into COM APIs and I _think_ those were resolved, but not able to directly find the answer.

Reply:
Since this isn't a static variable, it's effectively the same as placing it the constructor like it was previously.